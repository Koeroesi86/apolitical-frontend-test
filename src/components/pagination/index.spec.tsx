import React from 'react';
import {render, fireEvent} from '@testing-library/react';
import Pagination from './index';

describe('Pagination component', () => {
  it('should render', () => {
    const onPageChange = jest.fn();
    const result = render(
      <Pagination onPageChange={onPageChange} totalPages={10} page={2}/>
    );

    expect(result.asFragment()).toMatchSnapshot();
  });

  it('should dispatch page changes', async () => {
    const onPageChange = jest.fn();
    const result = render(
      <Pagination onPageChange={onPageChange} totalPages={10} page={2}/>
    );

    fireEvent.click(result.getByTestId('previous-page'));
    fireEvent.click(result.getByTestId('next-page'));
    fireEvent.click(result.getByTestId('page-8'));

    expect(onPageChange).toHaveBeenCalled();
    expect(onPageChange.mock.calls).toMatchSnapshot();
  });
});
