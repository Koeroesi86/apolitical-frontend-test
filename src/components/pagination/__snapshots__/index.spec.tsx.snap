// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component should dispatch page changes 1`] = `
Array [
  Array [
    1,
  ],
  Array [
    3,
  ],
  Array [
    8,
  ],
]
`;

exports[`Pagination component should render 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ gtvswt"
  >
    <button
      class="sc-gsTCUz gciomF"
      data-testid="previous-page"
    >
      Previous page
    </button>
    <button
      class="sc-dlfnbm YMeyH"
      data-testid="page-1"
    >
      1
    </button>
    <button
      class="sc-dlfnbm YMeyH"
      data-testid="page-2"
      disabled=""
    >
      2
    </button>
    <button
      class="sc-dlfnbm YMeyH"
      data-testid="page-3"
    >
      3
    </button>
    <button
      class="sc-dlfnbm YMeyH"
      data-testid="page-4"
    >
      4
    </button>
    <button
      class="sc-dlfnbm YMeyH"
      data-testid="page-5"
    >
      5
    </button>
    <button
      class="sc-dlfnbm YMeyH"
      data-testid="page-6"
    >
      6
    </button>
    <button
      class="sc-dlfnbm YMeyH"
      data-testid="page-7"
    >
      7
    </button>
    <button
      class="sc-dlfnbm YMeyH"
      data-testid="page-8"
    >
      8
    </button>
    <button
      class="sc-dlfnbm YMeyH"
      data-testid="page-9"
    >
      9
    </button>
    <button
      class="sc-dlfnbm YMeyH"
      data-testid="page-10"
    >
      10
    </button>
    <button
      class="sc-gsTCUz gciomF"
      data-testid="next-page"
    >
      Next page
    </button>
  </div>
</DocumentFragment>
`;
